packages/data-model/src/models/submissions.ts

```diff
@@ -1,22 +1,42 @@
 import { relations } from 'drizzle-orm';
 import { integer, jsonb, pgEnum, pgTable, serial, timestamp, varchar } from 'drizzle-orm/pg-core';
 
-import { SchemaData, SchemaValidationError } from '@overturebio-stack/lectern-client/lib/schema-entities.js';
+import {
+	type DataRecord,
+	SchemaData,
+	SchemaValidationError,
+} from '@overturebio-stack/lectern-client/lib/schema-entities.js';
 
 import { dictionaries } from './dictionaries.js';
 import { dictionaryCategories } from './dictionary_categories.js';
 
 export const submissionStatusEnum = pgEnum('submission_status', ['OPEN', 'VALID', 'INVALID', 'CLOSED', 'COMMITTED']);
 
-export type SubmissionData = {
+export type SubmissionInsertData = {
 	batchName: string;
 	creator: string;
 	records: SchemaData;
 };
 
+export type SubmissionUpdateData = {
+	systemId: string;
+	oldData: DataRecord;
+	newData: DataRecord;
+};
+
+export type SubmissionDeleteData = {
+	systemId: string;
+};
+
+export type SubmissionData = {
+	inserts?: Record<string, SubmissionInsertData>;
+	updates?: Record<string, SubmissionUpdateData[]>;
+	deletes?: Record<string, SubmissionDeleteData[]>;
+};
```

Comment:
This is the new structure of a Submission stored in the database.
Allowing to separate insert data, update and delete changes